### PR TITLE
Add rails 8  support

### PR DESCRIPTION
**Pull request by reeganviljoen**
### What are you trying to accomplish?

Add rails 8 appraisal and ci to matrix and move sprockets to be depended on for older versions

### CODE REVIEW

**Comment by reeganviljoen** [comment]
@joelhawksley I have left this out of the changelog as its more of a thing for developers on the project

**Comment by reeganviljoen** [comment]
@joelhawksley sorry I only see you also had a pr for this now, I have however fixed the issues that caused your pr to fail, if you don't mind taking a look

File: test/sandbox/config/application.rb [hunk 1]
```diff
@@ -7,7 +7,8 @@
 require "action_controller/railtie"
 require "action_mailer/railtie"
 require "action_view/railtie"
-require "sprockets/railtie"
+require "sprockets/railtie" if Rails.version.to_f <= 7.2 || !ENV["BENCHMARK"].present?
```
[the remark this answers is not in the document]
Comment (reeganviljoen) [reply test/sandbox/config/application.rb:10]: this is for the benchmark in ci wihch doesnt use appraiasal so I have to signal to not include sprockets somehow as its not in the gemspec anymore

File: test/sandbox/config/initializers/assets.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 # frozen_string_literal: true
 
-Rails.application.config.assets.precompile += %w[admin.css]
+if Rails.version.to_f <= 7.2 then Rails.application.config.assets.precompile += %w[admin.css] end
```
[the remark this answers is not in the document]
Comment (reeganviljoen) [reply test/sandbox/config/initializers/assets.rb:3]: yes but propshaft auto includes it

File: test/sandbox/test/rendering_test.rb [hunk 3]
```diff
@@ -330,16 +336,21 @@ def test_renders_component_without_format
...
-    assert_match(%r{http://assets.example.com/assets/application-\w+.css}, render_inline(component).text)
+      component.config.asset_host = "http://assets.example.com"
+      assert_match(%r{http://assets.example.com/assets/application-\w+.css}, render_inline(component).text)
 
-    component.config.asset_host = "assets.example.com"
-    assert_match(%r{http://assets.example.com/assets/application-\w+.css}, render_inline(component).text)
+      component.config.asset_host = "assets.example.com"
+      assert_match(%r{http://assets.example.com/assets/application-\w+.css}, render_inline(component).text)
+    else
+      assert_match(%r{http://assets.example.com/application.css}, render_inline(component).text)
```
[the remark this answers is not in the document]
Comment (reeganviljoen) [reply test/sandbox/test/rendering_test.rb]: I added propshaft it was a mistake :sweat:

**Comment by reeganviljoen** [comment]
@camertron I have added propshaft at last, it does however come with less features by design, like for instance I can not set a custom host as far as I know so those tests are omitted on new rails versions

File: test/sandbox/config/application.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (reeganviljoen) [reply test/sandbox/config/application.rb]: @camertron I removed this

**Comment by reeganviljoen** [comment]
@camertron @Spone @joelhawksley @BlakeWilliams  anyone had a chance to review this, it is quite important

**Comment by reeganviljoen** [comment]
Thanks @joelhawksley for the additions

File: test/sandbox/test/rendering_test.rb [hunk 4]
```diff
@@ -332,14 +336,19 @@ def test_renders_component_with_asset_url
     component = AssetComponent.new
     assert_match(%r{http://assets.example.com/assets/application-\w+.css}, render_inline(component).text)
 
-    component.config.asset_host = nil
-    assert_match(%r{/assets/application-\w+.css}, render_inline(component).text)
+    if Rails.version.to_f < 8.0
 
-    component.config.asset_host = "http://assets.example.com"
-    assert_match(%r{http://assets.example.com/assets/application-\w+.css}, render_inline(component).text)
+      #  Propshaft doesn't allow setting custom hosts so this only works in Rails < 8
```
Comment (joelhawksley) [inline test/sandbox/test/rendering_test.rb:341]:
```suggestion
      # Propshaft doesn't allow setting custom hosts so this only works in Rails < 8
```

**Pull request merged**